controllers/aggregated_clusteroperator_controller.go

```diff
@@ -62,34 +63,40 @@ func (a *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req
 	}
 	defer func() {
 		if err := coWriter.UpdateStatus(ctx, aggregatedCO, coBuilder.GetStatus()); err != nil {
-			log.Error(err, "error updating CO status")
+			log.Error(err, "error updating cluster operator status")
 		}
 	}()
 
 	// Set the default CO status conditions: Progressing=True, Degraded=False, Available=False
-	coBuilder.WithProgressing(openshiftconfigv1.ConditionTrue, "")
-	coBuilder.WithDegraded(openshiftconfigv1.ConditionFalse)
-	coBuilder.WithAvailable(openshiftconfigv1.ConditionFalse, "", "")
+	// TODO: always set a reason (message is optional, but desirable)
+	coBuilder.WithProgressing(metav1.ConditionTrue, "", "")
+	coBuilder.WithDegraded(metav1.ConditionFalse, "", "")
+	coBuilder.WithAvailable(metav1.ConditionFalse, "", "")
 
 	poList := &platformv1alpha1.PlatformOperatorList{}
 	if err := a.List(ctx, poList); err != nil {
 		return ctrl.Result{}, err
 	}
 	if len(poList.Items) == 0 {
 		// No POs on cluster, everything is fine
-		coBuilder.WithAvailable(openshiftconfigv1.ConditionTrue, "No POs are present in the cluster", "NoPOsFound")
-		coBuilder.WithProgressing(openshiftconfigv1.ConditionTrue, "No POs are present in the cluster")
+
+		// TODO: cleanup condition reasons.
+		//   1. use constants for condition reasons.
```

Comment:
We have some condition constants already defined but it doesn't seem like we are using them here for some reason. Maybe we just need to add onto that list?

https://github.com/openshift/platform-operators/blob/03149b26e54fa0737b37a9e183bb75fbf73ea94a/api/v1alpha1/types.go#L22-L25

Reply:
Those are reasons for the PO `Applied` condition type. These are reasons for the ClusterOperator condition types.